_source/_posts/2022-05-02-angular-microfrontend-auth.md

```diff
@@ -0,0 +1,443 @@
+  plugins: [
+    new ModuleFederationPlugin({
+        library: { type: "module" },
+
+        // For remotes (please adjust)
+        // name: "shell",
+        // filename: "remoteEntry.js",
+        // exposes: {
+        //     './Component': './projects/shell/src/app/app.component.ts',
+        // },        
+        
+        // For hosts (please adjust)
+        remotes: {
+          "mfeBasket": "http://localhost:4201/remoteEntry.js",
+        },
+
+        shared: share({
+          // ...important external libraries to share
+
+          ...sharedMappings.getDescriptors()
+        })
+        
+    }),
+    sharedMappings.getPlugin()
+  ],
+};
+
+```
+
+In the `webpack.config.js` for `mfe-basket`, you'll see the path for `@shared` at the top of the file and the configuration to identify what to expose in the remote application.
+
+The `shell` application serves on port 4200, and the `mfe-basket` application serves on port 4201. We can open up two terminals to run each application, or we can use the following npm script that the schematic to add `@angular-architects/module-federation` created for us:
+ 
+```shell
+npm run run:all
+``` 
+
+When you do so, you'll see both applications open in your browser and see how they fit together in the `shell` application running on port 4200. Click the **Basket** button to navigate to a new route that displays the `BasketModule` in the `mfe-basket` application. The sign-in button doesn't work quite yet, but we'll get it going here next.
+
+**Note** - _Another option I could have used for the starter is a [Nx workspace](https://nx.dev/). Nx has great tooling and built-in support for building micro-frontends with Webpack and Module Federation. But I wanted to go minimalistic on the project tooling so you'd have a chance to dip your toes into some of the configuration requirements._
+
+You might find the syntax `@shared` to look a little unusual since you may have expected to see a relative path to the library. The `@shared` syntax is an alias for the library's path, which is defined in the project's `tsconfig.json` file. You don't have to do this and can leave libraries using the relative path, but adding aliases makes your code look cleaner and helps ensure best practices for code architecture.
+
+Because the host application doesn't know about the remote applications except in the `webpack.config.js`, we help out the TypeScript compiler by declaring the remote application in `decl.d.ts`. You can see all the configuration changes and source code made for the starter [in this commit](https://github.com/oktadev/okta-angular-microfrontend-example/commit/b851320ca9418b8e28890fab15c7da5313eb8275).
+
+## Add authentication using Okta
+
+One of the most useful features of Module Federation is managing shared code and state. Let's see how this all works by adding authentication to the project and using the authenticated state in the existing application and a new micro-frontend. 
```

Comment:
```suggestion
One of the most useful features of Module Federation is managing shared code and state. Let's see how this all works by adding authentication to the project. We'll use the authenticated state in the existing application with a new micro-frontend. 
```
simplifying, but maybe i've changed the meaning by mistake.

Reply:
I updated to this (since the goal is to use the authenticated across 2 applications). Please LMK if there's any corrections to make:

> One of the most useful features of Module Federation is managing shared code and state. Let's see how this all works by adding authentication to the project. We'll use the authenticated state in the existing application and with a new micro frontend.